backend/src/tx.rs

```diff
@@ -0,0 +1,136 @@
+use subxt::ext::codec::Encode;
+use subxt::ext::sp_runtime::traits::{IdentifyAccount, Verify};
+use subxt::ext::{sp_core, sp_runtime};
+use subxt::{config::polkadot::PolkadotExtrinsicParams, config::Config};
+
+use crate::configuration::Configuration;
+use crate::error::AppError;
+use crate::utils::{calculate_signature, get_current_block, get_tx_counter};
+
+use self::kilt::runtime_types;
+use self::kilt::runtime_types::did::did_details::DidAuthorizedCallOperation;
+
+#[subxt::subxt(runtime_metadata_path = "./metadata.scale")]
+pub mod kilt {}
+
+pub type ProxyType = kilt::runtime_types::peregrine_runtime::ProxyType;
```

Comment:
I would rather use the spiritnet runtime. In the end it has to work with that.

Reply:
I was running into an error while using the metadata of spiritnet for peregrine. I can not exactly remember what the issue was, but I rather prefer to have both metadata and use them via a feature? [ae31e59](https://github.com/KILTprotocol/attester/pull/1/commits/ae31e59dc840a0de8e5a3337bd5800ceb06eb29d)